src/rand.rs

```diff
@@ -307,6 +313,47 @@ mod fuchsia {
     }
 }
 
+#[cfg(all(target_env = "sgx", target_feature = "rdrand"))]
+mod rdrandom {
+    use crate::error;
+    use crate::c;
+
+    #[must_use]
+    #[repr(transparent)]
+    pub struct OneMeansSuccess(c::int);
+
+    impl From<OneMeansSuccess> for Result<(), error::Unspecified> {
+        fn from(OneMeansSuccess(value): OneMeansSuccess) -> Self {
+            if value == 1 {
+                Ok(())
+            } else {
+                Err(error::Unspecified)
+            }
+        }
+    }
+
+    extern "C" {
+        fn CRYPTO_rdrand(out: &mut [u8; 8]) -> OneMeansSuccess;
+        // The |len| argument must be a multiple of eight.
+        fn CRYPTO_rdrand_multiple8_buf(buf: *mut u8, len: usize) -> OneMeansSuccess;
+    }
+
+    pub fn fill(dest: &mut [u8]) -> Result<(), error::Unspecified> {
+        unsafe {
+            let (prefix, rest) = dest.split_at_mut(dest.len() - (dest.len() % 8));
+            Result::from(CRYPTO_rdrand_multiple8_buf(prefix.as_mut_ptr(), prefix.len()))?;
+
+            if !rest.is_empty() {
+                let mut buf = [0; 8];
+                Result::from(CRYPTO_rdrand(&mut buf))?;
```

Comment:
I think it would be better to do everything with`CRYPTO_rdrand` and avoid `CRYPTO_rdrand_multiple8_buf`. Then we can implement the `retry 10 times` advice from Intel in this Rust code.

Reply:
Done